File: 백준/Bronze/2947. 나무 조각/나무 조각.c
#include <stdio.h>

int main(void) { 
   int arr[5], index = 0;

   for(int i = 0; i < 5; i++){
      scanf("%d", &arr[i]);      
   }

   while(1){
      if(arr[0] == 1 && arr[1] == 2 && arr[2] == 3 && arr[3] == 4 && arr[4] == 5){
         break;
      }

      if(arr[index] > arr[index + 1]){
         int temp = arr[index];
         arr[index] = arr[index + 1];
         arr[index + 1] = temp;

         for(int i = 0; i < 5; i++){
            printf("%d ", arr[i]);
         }
         printf("\n");
      }

      

      index += 1;
      if(index == 4){
         index = 0;
      }
   }

   return 0;
}

File: 백준/Bronze/2947. 나무 조각/나무 조각.py
arr = list(map(int, input().split()))
index = 0

while(True):
  if arr[0] == 1 and arr[1] == 2 and arr[2] == 3 and arr[3] == 4 and arr[4] == 5:
    break

  if arr[index] > arr[index + 1]:
    arr[index], arr[index + 1] = arr[index + 1], arr[index]

    for i in range(5):
      print(arr[i], end = ' ')
    
    print()

  index += 1
  if index >= 4:
    index = 0